boilermaker/task.py

```diff
@@ -40,6 +44,9 @@ def default(cls, function_name: str, **kwargs):
             **kwargs,
         )
 
+    def __bool__(self) -> bool:
+        return self.can_retry
```

Comment:
Seems like this could be useful but I don't know how I feel about overloading bool operator to know if it has retries left or not. It feels ambiguous and maybe arbitrary? We could have easily overloaded bool to return false if the function is empty string for example. Not sure when I'd ever use this operator though in general? I'd hate to see code like:

```
if task:
  return
```

vs. this code which is clearer

```
if task.can_retry:
  return
 ```

Reply:
I think I added this when trying to solve a bug... I agree it's a bit surprising.